1 CRUD Team

Velass opened the pull request.

Velass committed:
feat: finish CRUD Team

Velass committed:
Merge branch '1-création-dune-équipe-par-un-utilisateur' of https://github.com/2024-D10-groupe-7/api into 1-création-dune-équipe-par-un-utilisateur

lpreaux commented on src/main/java/fr/scrumtogether/scrumtogetherapi/exceptions/TeamException.java:
```diff
@@ -0,0 +1,12 @@
+package fr.scrumtogether.scrumtogetherapi.exceptions;
+
+public class TeamException extends RuntimeException {
```
J'aurais plutôt fait une EntityCreationException.
```suggestion
public class EntityCreationException extends RuntimeException {
```

lpreaux committed:
feat: enhance DTOs and mappers for Team and Project data

Introduces `TeamUserDto` and enhances `TeamDto` to include `ProjectDto` and user details. Adapts mappers and services to handle these new DTOs, ensuring streamlined data transformation for teams and their associated projects.

Velass committed:
feat: teamService finish
